enhancements/network/egress-qos.md

```diff
@@ -0,0 +1,107 @@
+---
+title: OVN Pods Egress DSCP QoS
+authors:
+  - "@oribon"
+reviewers:
+  - "@trozet"
+approvers:
+  - TBD
+creation-date: 2022-02-16
+last-updated: 2022-02-16
+status: implementable
+---
+
+# OVN Pods Egress DSCP QoS
+
+## Summary
+
+Not all traffic has the same priority, and when there is contention for bandwidth, there should be a mechanism for objects outside the cluster to prioritize the traffic.
+To enable this, we will use Differentiated Services Code Point (DSCP) which allows us to classify packets by setting a 6-bit field in the IP header, effectively marking the priority of a given packet relative to other packets as "Critical", "High Priority", "Best Effort" and so on.
+
+By introducing a new CRD `PodQos`, users could specify a DSCP value for packets originating from pods on a set of namespaces heading to a specified CIDR.
```

Comment:
If I was going to nitpick the name, it would be the "Pod" part more than the "QoS" part... (Isn't it more like "NamespaceQos"? or "NamespaceEgressQos"?)

Maybe as we nail down the use case (and possible future changes to it) some more a better name will become clear.

Reply:
I like "EgressQoS", just like we have egressIPs, egressFirewalls and maybe in the future egressGateways, fits into the picture since afterall it is QoS for egress traffic, restricting to Pod or Namespace at this point might hurt us in the future if we think of changing its scope?